pkgs/development/python-modules/dagger-io/default.nix

```
@@ -0,0 +1,60 @@
+{ lib
+, anyio
+, attrs
+, beartype
+, buildPythonPackage
+, cattrs
+, colorama
+, fetchFromGitHub
+, gql
+, graphql-core
+, httpx
+, platformdirs
+, poetry-core
+, pythonOlder
+, rich
+, shellingham
+, strawberry-graphql
+, typer
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "dagger-io";
+  version = "0.6.1";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.10";
+
+  src = fetchFromGitHub {
+    owner = "dagger";
+    repo = "dagger";
+    rev = "v${version}";
+    hash = "sha256-9QQ6aDCkTWNq5KOSGF6FH6UQrOYa51ctW3CMcGrCJAQ=";
+  } + "/sdk/python";
```

```suggestion
  inherit (dagger) src;
  
  sourceRoot = "source/sdk/python";
```
It seems there is already `dagger` in nixpkgs.
How about using its src?

---

I've no idea of how can I reference the source/sdk/python of another nixpkg.

The existing dagger package I think is the CLI tool. So I've assumed that I've to fetch from github the python sdk from the dagger-io repository to build the package. I just guided myself with other nixpkgs that does similar.

---

hnswlib should be helpful.
hnswlib is a C++ header library, but also provides a python module.
https://github.com/NixOS/nixpkgs/blob/8d56c1fe5a61939290694404947fa4127e7ced2f/pkgs/development/python-modules/hnswlib/default.nix
https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/libraries/hnswlib/default.nix